### DISCUSSION TITLE
Chlorinator status through API

### DISCUSSION THREAD

**Discussion by magimat** [tagyoureit/nodejs-poolController#445] [General]
I have an aquarite chlorinator that can report a few error status:  no flow, low salt, high salt and check cell

According to the wiki (https://github.com/tagyoureit/nodejs-poolController/wiki/Chlorinator) the protocol sends the info about this status, so I was hoping to be able to get this info through the API, but it doesn't seem to get passed through.

I noticed the status node in the json for the chlorinator info:

status: {
val: 0,
name: "ok",
desc: "Ok"
}

but the value never changes, even if the panel has a specific status led lit.

Is there any way to get this info through the API ??

thanks!

**Comment by rstrouse** [comment]
The api endpoint is GET /state/chlorinator/:id.  You can find the id of your chlorinator in the poolConfig.json.

So if the id is 1 then GET /state/chlorinator/1

**Comment by magimat** [comment]
Yea, got that already, but status info is not there

**Comment by rstrouse** [comment]
It should be in there.  I just checked the endpoint and it does report the status value.  Is it connected and communicating.  If it is not then perhaps that is the issue.
```
{
	"id": 1,
	"disabled": false,
	"lastComm": 1652726045775,
	"body": {
		"val": 0,
		"name": "pool",
		"desc": "Pool"
	},
	"targetOutput": 100,
	"saltLevel": 3000,
	"saltRequired": 113.33,
	"status": {
		"val": 0,
		"name": "ok",
		"desc": "Ok"
	},
	"currentOutput": 100,
	"type": {
		"val": 1,
		"name": "unknown",
		"desc": "unknown"
	},
	"name": "Intellichlor--40",
	"isActive": true,
	"superChlorRemaining": 0,
	"superChlor": false,
	"poolSetpoint": 100,
	"spaSetpoint": 0,
	"superChlorHours": 0,
	"model": {
		"val": 3,
		"name": "intellichlor--40",
		"desc": "IntelliChlor IC40",
		"capacity": 40000,
		"chlorinePerDay": 1.4,
		"chlorinePerSec": 0.000016203703703703704
	},
	"equipmentType": "chlorinator",
	"lockSetpoints": false
}
```

**Comment by magimat123** [comment]
yea, it does report the status value, I actually get something really similar to what you posted, but the status is always the same value regardless of any active indicator on the chlorinator itself

Maybe it's not being populated correclty in the code?

**Reply by magimat** [reply]
And by the way I can interact with the chlorinator and set the output and get salt level, so I know the software is communicating with my chlorinator just fine otherwise

**Comment by rstrouse** [comment]
That status is reporting what the RS485 reports.  Unfortunately, it looks like AquaRite doesn't report that data.

**Comment by magimat** [comment]
well, it appears your right :(

I did some debugging of my own and the status byte directly from the rs485 response indeed stays to 0x00 even where I stop the pump. Maybe it's just the no flow that's not being reported, I'll have to wait and see when I get any other status report on my chlorinator, unfortunately, LOW SALT, HIGH SALT and CHECK CELL is not easy to manually test, so I'll have to check whenever any of those occur..

thanks anyway

**Reply by rstrouse** [reply]
I think it depends on the age of the control box not necessarily the age of the cell as to how much info is reported through it.  You can at least monitor the salt levels but it is a real bummer that the flow switch seems to be an afterthought.

**Reply by magimat** [reply]
yea, I can set output % and get salt level, so that's already great
 
I already have access to another flow switch and can definitely infer low salt and high salt from the salt level. In fact, the only status I was really interested in was the inspect cell, because whenever it occurs, it can sometimes take a couple days before I notice, and meanwhile, it's not generating any chlorine, so I hoped to be able to hook it up to some notification through home assistant, but well. I will double check next time this status occurs, but I'm not holding my breath and it's not worth switching to a newer board just for that.

Just a bummer they didn't implement it, everything must be right there to do it!
